docs/guides/data-movement/copy-offload.md

```diff
@@ -0,0 +1,92 @@
+# Copy-Offload
+
+The copy-offload API allows a user's compute application to specify [Data Movement](../data-movement/readme.md) requests. The user's application utilizes the `libcopyoffload` library to establish a secure connection to the copy-offload server to initiate, list, query the status of, or cancel data movement requests. The copy-offload server accepts only those requests that present that Workflow's token.
+
+The copy-offload server is implemented as a special kind of [User Container](../user-containers/readme.md). Like all user containers, this is activated by a `DW container` directive in the user's job script and runs on the Rabbit nodes that are associated with the compute nodes in the user's job.
+
+## Administrative Configuration
+
+### TLS signing key and certificate
+
+A signing key and self-signed TLS certificate must be created and made available to the copy-offload server and the certificate must also be copied to each compute node. This certificate must have a SAN extension that describes all of the Rabbit nodes.
+
+Tools are available in the `nnf-dm` repository to assist in creating this certificate and its signing key. Begin by confirming that the cluster's `SystemConfiguration` resource can be accessed using the 'kubectl' command. This resource contains the information about all of the Rabbit nodes and is used when creating the SAN extension for the certificate:
+
+```console
+kubectl get systemconfiguration
+```
+
+From the `nnf-deploy` workarea, change into the `nnf-dm` directory and run `tools/mk-usercontainer-secrets.sh`:
+
+```console
+cd nnf-dm
+tools/mk-usercontainer-secrets.sh
+```
+
+That tool creates the signing key and the certificate and stores them in a Kubernetes secret named `nnf-dm-usercontainer-server-tls`. This first secret is mounted into the copy-offload server's pod when it is specified in a user's job script. The certificate is also stored alone in a Kubernetes secret named `nnf-dm-usercontainer-client-tls`. The content of this second secret can be retrieved by the administrator and copied to each compute node.
+
+```console
+CLIENT_TLS_SECRET=nnf-dm-usercontainer-client-tls
+kubectl get secrets $CLIENT_TLS_SECRET -o json | jq -rM '.data."tls.crt"' | base64 -d > cert.pem
+```
+
+> [!IMPORTANT]
+> Copy the certificate to `/etc/nnf-dm-usercontainer/cert.pem` on each compute node. It must be readable by all users' compute applications.
+
+### Rabbit hostname in /etc/hosts
+
+Each compute node must know the name of its matching rabbit node. This allows the `libcopyoffload` library, used in the user's compute application, to know which rabbit has the copy-offload server for that job. The library expects to use a generic name, `local-rabbit`, to contact that rabbit.
```

Comment:
This is going to change. I'm still waiting to see how the discussion in the slack thread settles.

Reply:
I assume that this could be applicable for user containers as well when there is a need for apps on the compute node to communicate with sockets-based entities running within a user container on the compute node

Reply:
Yes

Reply:
Marty, see https://cray.slack.com/archives/C020U81E05U/p1741714987737109

Reply:
Chris is adding an `/etc/local-rabbit.conf` file as discussed in the slack thread to our ansible config.  That makes sense to me for the gfs2 case, but for Lustre there's no guarantee Flux located any of the servers on the attached rabbit.  In this case I assume the rabbit just mounts the Lustre filesystem as needed?

Reply:
Blake has a PR out against the new API to address that. The compute app will have an environment variable, from the Workflow, telling it which rabbit to talk to.

Reply:
So in general, should the compute app always use the environment variable instead of what is in /etc/local-rabbit.conf?

Reply:
I wear both hats.  Under what conditions will one be present and not the other?